Source/GSStoryboardTransform.m

```diff
@@ -0,0 +1,871 @@
+{
+  return _initialViewControllerId;
+}
+
+- (NSString *) applicationSceneId
+{
+  return _applicationSceneId;
+}
+
+- (NSMapTable *) segueMapForIdentifier: (NSString *)identifier
+{
+  return [_identifierToSegueMap objectForKey: identifier];
+}
+
+- (NSXMLElement *) createCustomObjectWithId: (NSString *)ident
+                                   userLabel: (NSString *)userLabel
+                                 customClass: (NSString *)className
+{
+  NSXMLElement *customObject =
+    [[NSXMLElement alloc] initWithName: @"customObject"];
+  NSXMLNode *idValue =
+    [NSXMLNode attributeWithName: @"id"
+                     stringValue: ident];
+  NSXMLNode *usrLabel =
+    [NSXMLNode attributeWithName: @"userLabel"
+                     stringValue: userLabel];
+  NSXMLNode *customCls =
+    [NSXMLNode attributeWithName: @"customClass"
+                     stringValue: className];
+  
+  [customObject addAttribute: idValue];
+  [customObject addAttribute: usrLabel];
+  [customObject addAttribute: customCls];
+
+  AUTORELEASE(customObject);
+  
+  return customObject;
+}
+
+- (NSData *) dataForIdentifier: (NSString *)identifier
+{
+  NSString *sceneId = [_controllerMap objectForKey: identifier];
+  NSXMLDocument *xml = [_scenesMap objectForKey: sceneId];
+  return [xml XMLData];
+}
+
+- (NSData *) dataForApplicationScene
+{
+  return [self dataForIdentifier: APPLICATION];
+}
+
+- (void) processChild: (NSXMLElement *)child
+              withDoc: (NSXMLElement *)doc
+          withAppNode: (NSXMLNode *)appNode
+              sceneId: (NSString *)sceneId
+{
+  NSString *customClassString = nil;
+  if (appNode != nil)
+    {
+      NSXMLElement *objects = (NSXMLElement *)[appNode parent];
+      NSArray *appConsArr = [appNode nodesForXPath: @"connections" error: NULL];
+      NSXMLNode *appCons = [appConsArr objectAtIndex: 0];
+
+      if (appCons != nil)
+        {
+          [appCons detach];
+        }
+      
+      // Assign application scene...
+      ASSIGN(_applicationSceneId, sceneId);
+      [_controllerMap setObject: _applicationSceneId
+                         forKey: APPLICATION];
+      
+      // Move all application children to objects...
+      NSArray *appChildren = [appNode children];      
+      FOR_IN(NSXMLElement*, ae, appChildren)
+        {
+          [ae detach];
+          [objects addChild: ae];
+        }
+      END_FOR_IN(appChildren);
+      
+      // Remove the appNode
+      [appNode detach];
+      
+      // Add it to the document
+      [objects detach];
+      [doc addChild: objects];
+      
+      // create a customObject entry for NSApplication reference...
+      NSXMLNode *appCustomClass = (NSXMLNode *)[(NSXMLElement *)appNode
+                                                   attributeForName: @"customClass"];
+      customClassString = ([appCustomClass stringValue] == nil) ?
+        @"NSApplication" : [appCustomClass stringValue];
+      NSXMLElement *customObject = nil;
```

Comment:
As I previously explained you could extract this duplicated code into a helper method:

- (void) addStandardObjects: ( NSXMLElement *)objects classString: (NSString *) customClassString
 connections: (NSXMLNode *)appCons
{
NSXMLElement *customObject = nil;

       customObject = 
         [self createCustomObjectWithId: @"-3"
                              userLabel: @"Application"
                            customClass: @"NSObject"];
       [objects insertChild: customObject
                  atIndex: 0];
       customObject = 
         [self createCustomObjectWithId: @"-1"
                              userLabel: @"First Responder"
                            customClass: @"FirstResponder"];
       [objects insertChild: customObject
                  atIndex: 0];
       customObject =
         [self createCustomObjectWithId: @"-2"
                              userLabel: @"File's Owner"
                            customClass: customClassString];
       if (appCons != nil)
         {
           [customObject addChild: appCons];
         }
       [objects insertChild: customObject
                  atIndex: 0]; 
}

Calling that method from both branches would remove the code duplication.

Reply:
OMG, this is endless

Reply:
Done.